homeassistant/core.py

```diff
@@ -2340,6 +2342,7 @@ def async_set_internal(
                 EVENT_STATE_REPORTED,
                 {
                     "entity_id": entity_id,
+                    "last_reported": now,
```

Comment:
Should we prevent reading `last_reported` directly from a `State` which is still active (meaning the last_reported timestamp may be overwritten)?